pkg/sif/write.go

```diff
@@ -14,23 +14,29 @@ import (
 	"github.com/sylabs/sif/v2/pkg/sif"
 )
 
-// writeBlobToFileImage writes a blob to f.
-func (f *fileImage) writeBlobToFileImage(r io.Reader, rootIndex bool) error {
-	t := sif.DataOCIBlob
-	if rootIndex {
-		t = sif.DataOCIRootIndex
-	}
+// WriteBlob writes a blob to the SIF f, as a DataOCIBlob descriptor.
+func (f *OCIFileImage) WriteBlob(r io.Reader) error {
+	return f.writeBlob(r, sif.DataOCIBlob)
+}
+
+// writeRootIndex writes the RootIndex blob to the SIF f, as a
+// DataOCIRootIndex descriptor.
+func (f *OCIFileImage) writeRootIndex(r io.Reader) error {
+	return f.writeBlob(r, sif.DataOCIRootIndex)
+}
 
+func (f *OCIFileImage) writeBlob(r io.Reader, t sif.DataType) error {
 	di, err := sif.NewDescriptorInput(t, r)
 	if err != nil {
 		return err
 	}
 
-	return f.AddObject(di)
+	return f.sif.AddObject(di)
 }
 
-// writeIndexToSIF writes an image and all of its manifests and blobs to f.
-func (f *fileImage) writeImageToFileImage(img v1.Image) error {
+// WriteImage writes an image and all of its manifests and blobs to f. This
+// function does not update the RootIndex.
+func (f *OCIFileImage) WriteImage(img v1.Image) error {
```

Comment:
Wondering if we might want to hold off exposing this... at some point we likely want a `name.Reference` to be provided, which will be associated with the `Image`?

Reply:
I was keeping to the pattern of `layout.Path` here, where the `WriteXXXX` are 'dumb' functions that just write into the blobs folder, and don't touch the RootIndex. The `AppendXXXX` functions take options, so `AppendImage` in #83 would be the place that you'd specify a `name.Reference` that would become an annotation in the `RootIndex`.

If you'd prefer to deviate from the `Write` vs `Append` split as it is in ggcr layout.Path, I'm fine with that, though.

Reply:
Actually - perhaps holding of exposing this does make sense, as you say.... we talked a lot in the Update work  about avoiding 'holes' in the SIF where possible when the RootIndex has to be updated. Update was written and used in a way that makes sure the RootIndex is always at the end.

Will make this private for now.